[plugin.audio.pureradio] 3.0.0

bdommel opened the pull request:
Rewrite based on guidelines for addons for Kodi 19+

learningit commented on plugin.audio.pureradio/resources/lib/plugin.py:
```diff
@@ -0,0 +1,102 @@
+    def showWarning(self, message):
+        """Shows a popup window in the XBMC GUI for 5 seconds"""
+        self._log("showWarning message: %s" % message)
+        xbmc.executebuiltin('XBMC.Notification(Info:,' + message + ',6000)')
```
This is deprecated. It's better to  use xbmcgui.Dialog().notification()

bdommel replied:
replaced the depricated executebuiltin function for xbmcgui.Dialog().notification()

enen92 commented on plugin.audio.pureradio/resources/lib/addon.py:
```diff
@@ -0,0 +1,100 @@
+    params = p.get_plugin_parms()
+    action = params.get("action", '')
+    if action:
+        eval("%s(params)" % action)
```
You should avoid using eval calls, especially in entry points : https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html

bdommel replied:
replaced eval with static function calls

learningit commented on plugin.audio.pureradio/resources/lib/api.py:
```diff
@@ -0,0 +1,14 @@
+    return """http://pureradio.one:8000/"""
+
+def retrieve_podcasturl():
+    return """http://www.spreaker.com/show/1757189/episodes/feed"""
```
Are you sure this what you want here?

bdommel replied:
replace the static string with response from web server.

learningit commented on plugin.audio.pureradio/resources/lib/api.py:
```diff
@@ -0,0 +1,14 @@
+
+# functions with web interface
+def retrieve_streamurl():
+    return """http://pureradio.one:8000/"""
```
Are you sure about this return value?

bdommel replied:
replace the static string with response from web server.

learningit commented on plugin.audio.pureradio/resources/lib/plugin.py:
```diff
@@ -0,0 +1,102 @@
+        """Plays the media file pointed by the URL passed as argument."""
+        self._log("play_resolved_url pluginhandle = [%s] url = [%s]" % (self.pluginhandle, url))
+        listitem = xbmcgui.ListItem(path=url)
+        return xbmcplugin.setResolvedUrl(self.pluginhandle, True, listitem)
```
A general comment about some of your files. It's bad practice to not have an newline before the EOF.

bdommel replied:
added newline before EOF.

bdommel commented:
replaced eval with used static function calls 
replaced static urls with web response
replace depreciated notification with xbmcgui.Dialog().notification()
added newline at EOL

Thanks for the help!

bdommel commented:
The rebase did not go as expected. I will close this pull request and open a new one.
